src/shogun/kernel/ExponentialARDKernel.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2015 Wu Lin
+ * Written (W) 2012 Jacob Walker
```

Comment:
Do you want to write to Jacob and ask him whether he agrees to re-license his code under BSD?